env/src/lib.rs

```diff
@@ -283,6 +280,21 @@ impl LogBuilder {
 }
 
 impl Logger {
+    pub fn new() -> Logger {
+        let mut builder = LogBuilder::new();
+
+        if let Ok(s) = env::var("RUST_LOG") {
+            builder.parse(&s);
+        }
+
+        builder.build()
+    }
+
+    pub fn filter(&self) -> LogLevelFilter {
```

Comment:
Yeah it's a bit awkward. The split's there to avoid having to deal with
what happens if you log at the "off" level.
On Mon, Jun 27, 2016 at 12:30 PM Alex Crichton <notifications@github.com>
wrote:

> In env/src/lib.rs
> <https://github.com/rust-lang-nursery/log/pull/86#discussion_r68609741>:
>
> > @@ -283,6 +280,21 @@ impl LogBuilder {
> >  }
> >
> >  impl Logger {
> > +    pub fn new() -> Logger {
> > +        let mut builder = LogBuilder::new();
> > +
> > +        if let Ok(s) = env::var("RUST_LOG") {
> > +            builder.parse(&s);
> > +        }
> > +
> > +        builder.build()
> > +    }
> > +
> > +    pub fn filter(&self) -> LogLevelFilter {
>
> filter seems ok to me as the returned type is LogLevelFilter, but I've
> always been thrown a bit at least when working out the interaction between
> LogLevel, LogLevelFilter, and the various functions that consume them.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/rust-lang-nursery/log/pull/86/files/415469589ebd0d632758b19aae0d6de051b23ef5#r68609741>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABY2UbeM0Tj0tviqOvTDsqtQ1eyXWi1iks5qP_qbgaJpZM4I9CHs>
> .
>